docker/segment-anything/Dockerfile.exporter

```diff
@@ -9,13 +9,21 @@ ARG no_proxy
 RUN apt update && apt install -y git curl && \
     rm -rf /var/lib/apt/lists/*
 
-COPY requirements.txt /tmp/requirements.txt
-RUN pip install --no-cache-dir --require-hashes --no-deps -r /tmp/requirements.txt && rm /tmp/requirements.txt
+COPY requirements/* /tmp/requirements/
+
+# install pip-tools
+RUN pip install --require-hashes --no-deps -r /tmp/requirements/req-pin-deps.txt
 
 WORKDIR /ws
+
 RUN mkdir -p /ws/output
-RUN pip install -e git+https://github.com/facebookresearch/segment-anything/@6fdee8f2727f4506cfbbe553e23b895e27956588#egg=segment_anything
+# install segment-anything with pinned hash
+RUN pip install --no-deps -e git+https://github.com/facebookresearch/segment-anything/@6fdee8f2727f4506cfbbe553e23b895e27956588#egg=segment_anything
 WORKDIR /ws/src/segment-anything
+# generate requirements file for sagment-anything from setup.py with hashes
+RUN pip-compile --extra all,dev --generate-hashes --output-file=/tmp/requirements/requirements.txt
```

Comment:
Why are we doing this if we are going to create a pinned dependency list at runtime?

Reply:
@vinnamkim can you elaborate about the "runtime"? does it mean the docker image build time?

According to [this security risk reporting](https://github.com/openvinotoolkit/datumaro/security/code-scanning/13) by OSSF scorecard, we need to remedy `segment-anything` installation step in the `Dockerfile.exporter`. pip provides an option '--require-hashes' to the installation command for the secure installation but it cannot be used with editable mode that is used for installation of `segment-anything` module to the docker image.
The only way to resolve this issue is adding `--no-deps` to the `segment-anything` installation command with editable option now. if using `--no-deps` option, it will not install dependencies so i've added `pip-compile` stage to generate pinned dependencies for the `segment-anything` separately.

Reply:
> The only way to resolve this issue is adding --no-deps to the segment-anything installation command with editable option now. if using --no-deps option, it will not install dependencies so i've added pip-compile stage to generate pinned dependencies for the segment-anything separately.

AFAIK, the reason why we are doing this is "pinning" dependencies. However, you execute `pip-compile` at runtime. It means that the `requirements.txt` file is generated online (yes it includes the hash numbers but they are actually generated online). This seems that you are not actually "pinning" dependencies.

Reply:
@vinnamkim I thought those are the indirect dependencies to the docker image so i've pinned only the `segment-anything` with git commit hash. but as you pointed out, it's better to pinning all dependencies through pre-generated requirements file. I will upload new patch soon. thank you.